TestProject.Spec/StepDefinitions/DynamicControlsTestStepDefinitions.cs

```diff
@@ -0,0 +1,71 @@
+﻿using NUnit.Framework;
+using TestProject.Tests.Pages;
+using TestProject.Utils;
+
+namespace TestProject.Spec.StepDefinitions
+{
+    [Binding]
+    public class DynamicControlsTestStepDefinitions
+    {
+        private readonly ScenarioContext _scenarioContext;
+
+        public DynamicControlsTestStepDefinitions(ScenarioContext scenarioContext)
+        {
+            _scenarioContext = scenarioContext;
+        }
+
+        [BeforeScenario("UI")]
+        public static void Setup()
+        {
+            Browser.GetDriver().Navigate().GoToUrl(ConfigReader.GetConfigValue("webUIUrl"));
+        }
+
+        [AfterScenario("UI")]
+        public static void TearDown()
+        {
+            Browser.GetDriver().Quit();
+        }
+
+        [Given("I have opened the Dynamic Control page")]
+        public void GivenIHaveOpenedTheDynamicControlPage()
+        {
+            var mainPage = new MainPage();
+            mainPage.ClickOnDynamicControl();
+            _scenarioContext["mainPage"] = mainPage;
+        }
```

Comment:
I'd suggest moving those to other classes, as those are not unique for this test/page
UI tag methods should go to somehere like BrowserHooks class, and clicking on the dynamic control page should be a part of MainPageSteps. 
We're still following Single responsibility & page object, even though now it works with step definitions, so it's just another layer on top